handler.go

```diff
@@ -104,26 +104,22 @@ func (dh *DefaultHandler) AddRegistries(registries ...Registry) error {
 
 // Build retrieves the complete suite of functiosn and alias that has been configured
 // within this Handler. This handler is ready to be used with template engines
-// that accept FuncMap, such as html/template or text/template.
+// that accept FuncMap, such as html/template or text/template. It will also
+// cache the function map for future use to avoid rebuilding the function map
+// multiple times, so it is safe to call this method multiple times to retrieve
+// the same builded function map.
 //
 // NOTE: This will replace the `FuncsMap()`, `TxtFuncMap()` and `HtmlFuncMap()` from sprig
 func (dh *DefaultHandler) Build() FunctionMap {
+	if dh.built {
+		return dh.cachedFuncsMap
+	}
+
 	AssignAliases(dh) // Ensure all aliases are processed before returning the registry
 	AssignNotices(dh) // Ensure all notices are processed before returning the registry
```

Comment:
`AssignAliases` and `AssignNotices` are bound to the `Handler` interface, not directly to the `DefaultHandler`. This allows other handlers to use the same feature without needing to re-implement it :)

Reply:
I change the behavior of `buildSafeFuncs` to follow the same idea / convention and allow to use it externally with `AssignSafeFuncs(handler Handler)` instead of be dependent of DefaultHandler